op-service/log/dynamic.go

```diff
@@ -12,28 +12,42 @@ type LvlSetter interface {
 
 // DynamicLogHandler allow runtime-configuration of the log handler.
 type DynamicLogHandler struct {
-	slog.Handler // embedded, to expose any extra methods the underlying handler might provide
-	minLvl       slog.Level
+	h      slog.Handler
+	minLvl *slog.Level // shared with derived dynamic handlers
 }
 
 func NewDynamicLogHandler(lvl slog.Level, h slog.Handler) *DynamicLogHandler {
 	return &DynamicLogHandler{
-		Handler: h,
-		minLvl:  lvl,
+		h:      h,
+		minLvl: &lvl,
 	}
 }
 
 func (d *DynamicLogHandler) SetLogLevel(lvl slog.Level) {
-	d.minLvl = lvl
+	*d.minLvl = lvl
```

Comment:
Do we need to make this atomic / add a lock if this is multithreaded?

Reply:
It would indeed be cleaner to make this atomic, will do

Reply:
We only re-set it to a different value in the admin rpc, which is super infrequent. And it's not critical if something was logging at a wrong level for a ms. So we decided to keep it as it is.